pageserver/src/page_service.rs

```diff
@@ -735,24 +766,40 @@ impl PageServerHandler {
         }))
     }
 
-    fn handle_get_page_at_lsn_request<T: DatadirTimeline>(
+    fn handle_get_page_at_lsn_request(
         &self,
-        timeline: &T,
+        timeline: &Arc<LayeredTimeline>,
         req: &PagestreamGetPageRequest,
     ) -> Result<PagestreamBeMessage> {
         let _enter = info_span!("get_page", rel = %req.rel, blkno = &req.blkno, req_lsn = %req.lsn)
             .entered();
         let latest_gc_cutoff_lsn = timeline.get_latest_gc_cutoff_lsn();
-        let lsn = Self::wait_or_get_last_lsn(timeline, req.lsn, req.latest, &latest_gc_cutoff_lsn)?;
+        let lsn = Self::wait_or_get_last_lsn(
+            timeline.as_ref(),
+            req.lsn,
+            req.latest,
+            &latest_gc_cutoff_lsn,
+        )?;
         /*
         // Add a 1s delay to some requests. The delayed causes the requests to
         // hit the race condition from github issue #1047 more easily.
         use rand::Rng;
         if rand::thread_rng().gen::<u8>() < 5 {
             std::thread::sleep(std::time::Duration::from_millis(1000));
         }
-        */
-        let page = timeline.get_rel_page_at_lsn(req.rel, req.blkno, lsn)?;
+         */
+
+        let page = tokio::runtime::Handle::current().block_on(async {
```

Comment:
Any reasons to avoid passing runtime directly as an argument? So there is no need to obtain a handle on each getpage request